sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullExpressions.scala

```diff
@@ -321,7 +322,12 @@ case class IsNull(child: Expression) extends UnaryExpression with Predicate {
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val eval = child.genCode(ctx)
-    ExprCode(code = eval.code, isNull = "false", value = eval.isNull)
+    val value = if ("true" == s"${eval.isNull}" || "false" == s"${eval.isNull}") {
```

Comment:
`if ("true" == eval.isNull || "false" == eval.isNull) {`?

Reply:
or `eval.isNull == Literal("true")`? Or even better we can create a `LiteralTrue = Literal("true")` and equivalent for false and use them?

Reply:
We can do `eval.isNull.instanceOf[LiteralValue]` as suggested by @cloud-fan below.